strawberry/annotation.py

```diff
@@ -63,14 +67,36 @@ def __eq__(self, other: object) -> bool:
 
         return self.resolve() == other.resolve()
 
+    def _parse_annotated(self, annotation: Type) -> object:
+        if get_origin(annotation) is Annotated:
+            annotated_args = get_args(annotation)
+
+            annotation_type = annotated_args[0]
+
+            for arg in annotated_args[1:]:
+                if isinstance(arg, StrawberryLazyReference):
+                    assert isinstance(annotation_type, ForwardRef)
+
+                    return arg.resolve_forward_ref(annotation_type)
+
+        if is_union(annotation):
+            return Union[
+                tuple(self._parse_annotated(arg) for arg in get_args(annotation))
+            ]
+
+        if is_list(annotation):
+            return List[self._parse_annotated(get_args(annotation)[0])]
+
+        return annotation
+
     def resolve(self) -> Union[StrawberryType, type]:
-        annotation: object
+        annotation = self._parse_annotated(self.annotation)
+
```

Comment:
@BryceBeagle does this make sense to you? I'm recursively parsing the annotations that use `Annotated`

I'll try to add some unit tests later